primary/src/certificate_waiter.rs

```diff
@@ -44,8 +41,8 @@ pub struct CertificateWaiter<PublicKey: VerifyingKey> {
     committee: Committee<PublicKey>,
     /// The persistent storage.
     store: Store<CertificateDigest, Certificate<PublicKey>>,
-    /// The current consensus round (used for cleanup).
-    consensus_round: Arc<AtomicU64>,
+    /// Receiver for signal of round change
+    rx_new_consensus_round: watch::Receiver<u64>,
```

Comment:
Nit: to be honest I would probably stick to `rx_consensus_round` or `rx_current_consensus_round` . Whether the value communicated through is new or not is another matter and something that we detect via the `changed` method anyways. But again, a nit :)

Reply:
I would go for `rx_consensus_round_updates`. A comment on this to indicate what the senders would look like would help.